From 4d2ad93b5ca1999be22876a88b7b4282b79fe439 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Mon, 17 Feb 2020 18:04:29 +0800
Subject: [PATCH] dt-bindings: can: rockchip_can: add can support for rockchip

This patch add the rockchip can to the device tree bindings
documentation.

Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
Change-Id: I7a15174dda4a6eec6196fca5e2c386a7844b70ee
---
 .../bindings/net/can/rockchip_can.txt         | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/can/rockchip_can.txt

diff --git a/Documentation/devicetree/bindings/net/can/rockchip_can.txt b/Documentation/devicetree/bindings/net/can/rockchip_can.txt
new file mode 100644
index 000000000000..72a22f2754d4
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/can/rockchip_can.txt
@@ -0,0 +1,32 @@
+Rockchip CAN controller Device Tree Bindings
+---------------------------------------------------------
+
+Required properties:
+- compatible		: Should be:
+			  - "rockchip,can-1.0" for CAN controllers 1.0
+- reg			: Physical base address and size of the controller
+			  registers map.
+- interrupts		: Property with a value describing the interrupt
+			  number.
+- clock-names		: List of input clock names
+			  - "can_clk", "pclk",
+			  (See clock bindings for details).
+- clocks		: Clock phandles (see clock bindings for details).
+- resets : Must contain an entry for each entry in reset-names.
+	   See ../reset/reset.txt for details.
+- reset-names : List of input reset names
+			  - "can", "can-apb".
+
+Example:
+
+For Dts file:
+	can: can@ff250000 {
+		compatible = "rockchip,can-1.0";
+		reg = <0x0 0xff610000 0x0 0x1000>;
+		interrupts = <GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru CLK_CAN>, <&cru PCLK_CAN>;
+		clock-names = "baudclk", "apb_pclk";
+		resets = <&cru SRST_CAN>, &cru SRST_CAN_P>;
+		reset-names = "can", "can-apb";
+		status = "okay";
+	};
-- 
2.35.3

